cadence/contracts/bridge/FlowEVMBridgeTemplates.cdc

```diff
@@ -0,0 +1,80 @@
+    /// Canonical path for the Admin resource
+    access(all) let AdminStoragePath: StoragePath
+    /// Chunked Hex-encoded Cadence contract code, to be joined on derived contract name
+    access(self) let templateCodeChunks: {String: [String]}
+
+    /// Serves Locker contract code for a given type, deriving the contract name from the type identifier
+    access(all) fun getLockerContractCode(forType: Type): [UInt8]? {
+        if forType.isSubtype(of: Type<@{NonFungibleToken.NFT}>()) && !forType.isSubtype(of: Type<@{FungibleToken.Vault}>()) {
+            return self.getNFTLockerContractCode(forType: forType)
+        } else if !forType.isSubtype(of: Type<@{FungibleToken.Vault}>()) && forType.isSubtype(of: Type<@{FungibleToken.Vault}>()) {
+            // TODO
+            return nil
+        }
+        return nil
+    }
+
+    /// Serves bridged asset contract code for a given type, deriving the contract name from the EVM contract info
+    // TODO: Consider adding the values we would need to derive from instead of abstracting EVM calls in scope
+    // access(all) fun getBridgedAssetContractCode(forEVMContract: EVM.EVMAddress): [UInt8]? {}
+
+    access(self) fun getNFTLockerContractCode(forType: Type): [UInt8]? {
+        if let contractName: String = FlowEVMBridgeUtils.deriveLockerContractName(fromType: forType) {
+            let contraNameHex: String = String.encodeHex(contractName.utf8)
+
+            // Construct the contract code from the templated chunked contract hex
+            let code: [UInt8] = []
+            let chunks: [String] = self.templateCodeChunks["nftLocker"]!
+            for i, chunk in chunks {
+                code.appendAll(chunk.decodeHex())
+                // No need to append the contract name after the last chunk
+                if i == chunks.length - 1 {
+                    break
+                }
+                code.appendAll(contraNameHex.decodeHex())
+            }
+            return code
+        }
+
+        return nil
+    }
+
+    /// Resource enabling updates to the contract template code
+    access(all) resource Admin {
+        access(all) fun upsertContractCodeChunks(forTemplate: String, chunks: [String]) {
+            FlowEVMBridgeTemplates.templateCodeChunks[forTemplate] = chunks
+        }
+        access(all) fun addNewContractCodeChunks(newTemplate: String, chunks: [String]) {
+            pre {
+                FlowEVMBridgeTemplates.templateCodeChunks[newTemplate] == nil: "Code already exists for template"
+            }
+            FlowEVMBridgeTemplates.templateCodeChunks[newTemplate] = chunks
+        }
+    }
+
+
+    // Flow CLI currently breaks flow.json on [String] in contract init - hard coding for the time being but needs new
+    // hex on template changes
+    // init(templateCodeChunks: {String: [String]}) {
+    init() {
+        self.AdminStoragePath = StoragePath(
+                identifier: "flowEVMBridgeTemplatesAdmin_".concat(self.account.address.toString())
```

Comment:
There will only be one of these contracts deployed to the bridge account. Guess it's a force of habit to prevent path conflicts, but not really needed given the single purpose of the bridge account